homework/index.js

```diff
@@ -29,41 +29,37 @@
     return elem;
   }
 
-  function makeText(stringDateAndTime) {
+  function addRepoDetailsInfo(listElm, title, value, link) {
+    const trElm = createAndAppend('tr', listElm);
+    createAndAppend('td', trElm, {
+      class: 'titles',
+      text: title,
+    });
+    const secondTdElm = createAndAppend('td', trElm, {
+      class: 'values',
+    });
+
+    if (link) {
+      createAndAppend('a', secondTdElm, { href: link, text: value });
+    } else {
+      secondTdElm.innerText = value;
+    }
+
+    return trElm;
+  }
```

Comment:
This is a nice improvement which makes your code DRY. An alternative to adding a fourth parameter for the link and using an `if` statement inside the function body is to take the exception for the first row out of this function and make it a responsibility of the caller. That was in my hint in my initial feedback. Your function could then simply look like this:

```js
function addRepoDetailsInfo(listElm, title, value, link) {
  const trElm = createAndAppend('tr', listElm);
  createAndAppend('td', trElm, {
    class: 'titles',
    text: title,
  });
  createAndAppend('td', trElm, {
    class: 'values',
    text: value,
  });
  return trElm;
}
```

And you would call it like this:

```js
const firstRow = addRepoDetailsInfo(table, 'Repository :', '');
createAndAppend('a', firstRow.lastChild, {
  text: repo.name,
  href: repo.html_url,
});
addRepoDetailsInfo(table, 'Description :', repo.description);
addRepoDetailsInfo(table, 'Forks :', repo.forks);
addRepoDetailsInfo(table, 'Updated :', formatDate(repo.updated_at));
```

Having said that, your solution looks fine too.